docs/dev/getting_started.rst

```diff
@@ -1,116 +1,145 @@
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-You'll need to install the following dependencies:
+#. Install `Python <https://www.python.org/downloads/windows/>`_ if you are on Windows, on Linux and OSX Python is preinstalled (recommended versions 2.7+ or 3.4+).
+#. Install `pip <https://pypi.python.org/pypi/pip>`_ package installer.
+#. Install `Node <https://nodejs.org/en/>`_ (recommended version 4+).
+   
+   .. note::
+     * On Ubuntu install Node.js via `nvm <https://github.com/creationix/nvm>`_ to avoid build issues.
+     * On a Mac, you may want to consider using the `Homebrew <http://brew.sh/>`_ package manager.
 
-- Python (including pip) - recommended version 2.7+ or 3.4+
-- Node.js - recommended version 4+
-- git
+Ready for the fun part in the Terminal? Here we go!
 
-The process for installing these depends on your operating system.
 
-.. note::
+Checking out the code
+~~~~~~~~~~~~~~~~~~~~~
 
-  - On Ubuntu, it is recommended to install Node.js via `nvm <https://github.com/creationix/nvm>`_ to avoid build issues.
-  - On a Mac, you may want to consider using the `Homebrew <http://brew.sh/>`_ package manager.
+#. Make sure you `registered your SSH keys on GitHub <https://help.github.com/articles/generating-ssh-keys>`_.
+#. **Clone** your Kolibri fork to your local computer. In the following commands replace ``$USERNAME`` with your own GitHub username:
 
+   .. code-block:: bash
 
+      # using SSH
+      git clone git@github.com:$USERNAME/kolibri.git
+      # using HTTPS
+      git clone https://github.com/$USERNAME/kolibri.git
 
-Clone the Repository
-~~~~~~~~~~~~~~~~~~~~
+#. Enable syncing your local repository with **upstream**,  which refers to the Kolibri source from where you cloned your fork. That way you can keep it updated with the changes from the rest of Kolibri team contributors:
 
-First clone the repo:
+  .. code-block:: bash
 
-.. code-block:: bash
+    cd kolibri  # Change into the newly cloned directory
+    git remote add upstream git@github.com:learningequality/kolibri.git  # Add the upstream
+    git fetch  # Check if there are changes upstream
+    git checkout develop
 
-  git clone git@github.com:learningequality/kolibri.git
+.. warning::
+  ``develop`` is the active development branch - do not target the ``master`` branch.
 
-Then, ``cd`` into the new directory:
 
-.. code-block:: bash
+Virtual environment
+~~~~~~~~~~~~~~~~~~~
 
-  cd kolibri
+It's generally good practice to use `Python virtual environment <https://virtualenv.pypa.io/en/latest/>`_ to isolate your project(s) and allow you to install different packages, even with different versions. This also allows you to avoid using ``sudo`` with the ``pip`` commands below, which is not recommended.
 
+Follow these `instructions to create a new virtual environment and activate it <http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_. You may also want to consider using `virtualenvwrapper <http://virtualenvwrapper.readthedocs.io/en/latest/index.html>`_.
 
```

Comment:
Personally, I avoid `virtualenvwrapper` (and use vanilla `virtualenv`) because it doesn't play nicely with my shell of choice (Fish). On one hand, the wrapper is a convenience for many people, but on the other hand these instructions might trip up folks who can't use the weapper.